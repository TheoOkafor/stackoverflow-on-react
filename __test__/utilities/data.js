const data = {
  answer: {
    accepted: null,
    body: 'What?',
    comments: [
      {
        answerid: 17,
        body: 'Jonzing',
        id: 30,
        questionid: 4,
        timesubmitted: '2018-11-21T12:35:08.000Z',
        userid: 4,
        username: 'teddy01'
      },
    ],
    downvotes: 0,
    id: 74,
    questionid: 38,
    timesubmitted: '2018-11-21T18:20:11.000Z',
    upvotes: 0,
    userid: 4,
    username: 'teddy01',
    votesCount: 0,
  },
  comment: {
    answerid: 17,
    body: 'Jonzing',
    id: 30,
    questionid: 4,
    timesubmitted: '2018-11-21T12:35:08.000Z',
    userid: 4,
    username: 'teddy01'
  },
  sampleAnswer: {
    data: {
      body: 'Jonzing',
      questionid: 4,
      timesubmitted: '2018-11-21T12:35:08.000Z',
      userid: 4,
      username: 'teddy01'
    },
  },
  question: {
    data: {
      answers: [
        {
          accepted: false,
          body: 'Lorem ipsum dolor sit amet, ',
          comments: [],
          downvotes: 0,
          id: 4,
          questionid: 4,
          timesubmitted: '2018-08-22T11:13:09.000Z',
          upvotes: 1,
          userid: 3,
          username: 'chanpion',
          votesCount: 1
        },
        {
          accepted: false,
          body: 'Lorem ipsum dolor sit amet, ',
          comments: [],
          downvotes: 0,
          id: 8,
          questionid: 4,
          timesubmitted: '2018-08-22T11:13:09.000Z',
          upvotes: 0,
          userid: 3,
          username: 'chanpion',
          votesCount: 0
        },
        {
          accepted: false,
          body: 'Lorem ipsum dolor sit amet, ',
          comments: [],
          downvotes: 0,
          id: 12,
          questionid: 4,
          timesubmitted: '2018-08-22T11:13:09.000Z',
          upvotes: 0,
          userid: 3,
          username: 'chanpion',
          votesCount: 0
        },
        {
          accepted: false,
          body: 'Lorem ipsum dolor sit amet, ',
          comments: [],
          downvotes: 0,
          id: 16,
          questionid: 4,
          timesubmitted: '2018-08-22T11:13:09.000Z',
          upvotes: 0,
          userid: 3,
          username: 'chanpion',
          votesCount: 0
        },
        {
          accepted: false,
          body: 'Lorem ipsum dolor sit amet, ',
          comments: [{
            answerid: 17,
            body: 'Jonzing',
            id: 30,
            questionid: 4,
            timesubmitted: '2018-11-21T12:35:08.000Z',
            userid: 4,
            username: 'teddy01'
          }],
          downvotes: 0,
          id: 17,
          questionid: 4,
          timesubmitted: '2018-08-22T11:13:09.000Z',
          upvotes: 0,
          userid: 3,
          username: 'chanpion',
          votesCount: 0
        },
        {
          accepted: null,
          body: 'worddy',
          comments: [],
          downvotes: 0,
          id: 67,
          questionid: 4,
          timesubmitted: '2018-11-21T12:23:12.000Z',
          upvotes: 0,
          userid: 4,
          username: 'teddy01',
          votesCount: 0
        }
      ],
      body: 'Lorem ipsum dolor sit amet, ',
      id: 4,
      timesubmitted: '2018-08-22T11:13:09.000Z',
      title: 'How can one learn it all so fast?',
      userid: 3,
      username: 'chanpion'
    },
    message: 'Question 4 found',
    statusCode: 200
  },
  questions: [
    {
      id: 1,
      title: 'why do yo shake?',
      body: 'Lorem ipsum dolor sit amet, ',
      userid: 3,
      username: 'chanpion',
      answers: [],
    },
    {
      id: 2,
      title: 'why do yo shake?',
      body: 'Lorem ipsum dolor sit amet, ',
      userid: 3,
      username: 'chanpion',
      answers: [],
    },
    {
      id: 3,
      title: 'why do yo shake?',
      body: 'Lorem ipsum dolor sit amet, ',
      userid: 3,
      username: 'chanpion',
      answers: [],
    },
  ],
  newQuestion: {
    id: 1,
    title: 'why do yo shake?',
    body: 'Lorem ipsum dolor sit amet, ',
    userid: 3,
    username: 'chanpion',
    answers: [],
  },
  payload: {
    username: 'Xtreme',
    'x-access-token': 'token',
  },
  userData: {
    firstName: 'Xtreme',
    lastName: 'Prodigy',
    username: 'Xtreme',
    password: 'password',
    confirmpassword: 'password',
  },
};

export default data;
