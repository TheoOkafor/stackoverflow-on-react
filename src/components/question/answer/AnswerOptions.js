import React from 'react';
import PropTypes from 'prop-types';
import Button from '../../main/Button';

const AnswerOptions = (props) => {
  return (
    <div className="answer-options">
      <div className="message-display">
        <p id={`"vote-mssg-${props.answer.id}"`} />
        <div
          className="loader-xs"
          id={`"vote-${props.answer.id}-loader"`} />
      </div>
      <span className="upvote-display">
        {props.answer.upvotes}
      </span>
      <Button
        styleName="primary-o"
        id={`"vote-${props.answer.id}-up"`}
        name="Upvote" />

      <span className="vote-display">
        {props.answer.downvotes}
      </span>

      <Button
        styleName="link gray"
        id={`"vote-${props.answer.id}-down"`}
        name="Downvote" />
      <Button
        styleName="link"
        value={`"${props.index}"`}
        name="Comment" />
    </div>
  );
};

AnswerOptions.propTypes = {
  id: PropTypes.string,
  index: PropTypes.number,
  // status: PropTypes.boolean,
  answer: PropTypes.object,
};

export default AnswerOptions;
