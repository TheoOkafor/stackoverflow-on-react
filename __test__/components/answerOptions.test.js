import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, fireEvent, render } from 'react-testing-library';
import { MemoryRouter } from 'react-router-dom';
import AnswerOptions from '../../src/components/question/answer/AnswerOptions';
import data from '../utilities/data';

afterEach(cleanup);
describe('AnswerOptions component', () => {
  test('Test that the AnswerOptions displays', () => {
    const component = renderer.create(
      <AnswerOptions answer={data.answer} index={12}/>
    );
    expect(component).toMatchSnapshot();
  });

  test('should handle click for upvoting answer', () => {
    const handleClick = jest.fn();
    const { getByTestId } = render(
      <MemoryRouter>
        <AnswerOptions
          answer={data.answer}
          index={12}
          handleVoting={jest.fn()} />
      </MemoryRouter>
    );
    const upvoteBtn = getByTestId('upvote');
    upvoteBtn.onclick = handleClick;
    fireEvent.click(upvoteBtn);

    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
