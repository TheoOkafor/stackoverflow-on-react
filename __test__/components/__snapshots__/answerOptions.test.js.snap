// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test that the AnswerOptions displays 1`] = `
<div
  className="answer-options"
>
  <div
    className="message-display"
  >
    <p
      id="\\"vote-mssg-74\\""
    />
    <div
      className="loader-xs"
      id="\\"vote-74-loader\\""
    />
  </div>
  <span
    className="upvote-display"
  >
    0
  </span>
  <button
    className="btn primary-o"
    id="\\"vote-74-up\\""
  >
    Upvote
  </button>
  <span
    className="vote-display"
  >
    0
  </span>
  <button
    className="btn link gray"
    id="\\"vote-74-down\\""
  >
    Downvote
  </button>
  <button
    className="btn link"
    value="\\"12\\""
  >
    Comment
  </button>
</div>
`;
